packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -130,49 +123,63 @@ function SelectionTool(props: Props) {
     'Escape',
     () => {
       startEvtRef.current = undefined;
-      setSelectionState({ selection: undefined, isCancelled: true });
+      setRawSelection(undefined);
     },
     [],
     { event: 'keydown' }
   );
 
-  useEffect(() => {
-    const { selection, isCancelled } = selectionState;
+  // Compute effective selection
+  const selection = useMemo(
+    () => rawSelection && transformSelectionRef.current(rawSelection),
+    [rawSelection, transformSelectionRef]
+  );
```

Comment:
The transformed/effective selection is now computed as a derived state, during render. It's memoised, of course, so as to not trigger the effect unnecessarily.